backup/backup.go

```diff
@@ -79,19 +138,45 @@ func (b *BackupManager) convertEtcdToBackup(resp []*mvccpb.KeyValue) backup {
 	return values
 }
 
-func (f backup) writeTo(filename string) error {
-	file, err := json.Marshal(f)
+func (b *BackupManager) loadSnapshot(filename string) error {
+	backupData := backup{}
+	err := backupData.readFrom(filename)
 	if err != nil {
-		return fmt.Errorf("could not marshal snapshot: %v", err)
+		return fmt.Errorf("could not read from file %s: %v", filename, err)
 	}
-	if err = ioutil.WriteFile(filename, file, 0644); err != nil {
-		return fmt.Errorf("could not write snapshot to file: %v", err)
+	fmt.Println("Clearing ETCD")
+	err = b.clearEtcd()
+	if err != nil {
+		return fmt.Errorf("could not clear ETCD: %v", err)
+	}
+	fmt.Println("Writing to ETCD")
+	err = b.writeToEtcd(backupData)
+	if err != nil {
+		return fmt.Errorf("could not write to ETCD: %v", err)
+	}
+	return nil
+}
+
+func (b *BackupManager) writeToEtcd(data backup) error {
+	for _, row := range data {
+		_, err := b.ETCDClient.Put(context.Background(), string(row.Key), string(row.Value))
+		if err != nil {
```

Comment:
Does `ETCDClient.Put` have retries built in?

Reply:
I believe it has built in retries but times out after a time limit. It could result in a partial restore. 

I can make it so that if a key fails, it clears etcd again

Reply:
I'm not sure whats best here, probably have it try its best then make sure its very clear that it only partially restored.